### PR TITLE
Exclude vendor from linting

### PR DESCRIPTION
**Pull request by mdhorn**
Changes proposed in this pull request:
- Exclude the vendor directory from lint commands
- Added new target for lint including vendor for release testing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -212,7 +212,29 @@ lint: build install-linters gopath
 	@rm -rf ${LOCAL_GOPATH}/src/${GO_PACKAGE_PREFIX}/vendor
 	@cp -af vendor/* ${LOCAL_GOPATH}/src/
 	@go build -race github.com/clearlinux/clr-installer/...
-	@gometalinter.v2 --deadline=10m --tests --vendor \
+	@gometalinter.v2 --deadline=10m --tests --vendor --skip=vendor \
+	--exclude=vendor --disable-all \
```
Comment (reaganlo) [inline Makefile]: Do we need this `--exclude=vendor --disable-all \` if we are already skipping vendor `--skip=vendor`?

Comment (mdhorn) [reply Makefile]: exlcude Exclude messages matching these regular expressions
but skip, Skip directories with this name when expanding '...'.

So the skip should shorten the processing time by never running on the vendor code .

File: .travis.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ script:
     - docker exec -it clear-test bash -c "cd /travis ; make dist-clean"
     - docker exec -it clear-test bash -c "cd /travis ; make"
     - docker exec -it clear-test bash -c "cd /travis ; PATH=${PATH}:${GOPATH}:/go/bin make vendor-check"
-    - docker exec -it clear-test bash -c "cd /travis ; make lint"
+    - travis_wait 30 sleep infinity & docker exec -it clear-test bash -c "cd /travis ; make lint"
```
Comment (reaganlo) [inline .travis.yml]: So we are keeping lint deadline as 20m and travis_wait as 30m? Just checking if this is intentional.

Comment (mdhorn) [reply .travis.yml]: No, they should be consistent.

**Pull request merged**